### PR TITLE
Standardize Global Parameter Definitions

### LINKED ISSUE #656: Standardize params.Config in Beacon-Chain and Validator Projects
Hi all,

We currently have two different ways of getting configuration params in both projects, the beacon chain uses params.GetConfig() while the validator uses params.DemoConfig() or params.DefaultConfig(). This is a bit confusing and would be best if we just stick to the way the beacon chain does so. This obfuscates what type of config parameters the system is using and lets the cli and node handle this effectively through a flag. This is preferred.

A PR to close this would change validator's params package to match the way the beacon-chain handles its params.

### PR DESCRIPTION
**Pull request by cyberbono3**
Resolves #656

Description
Standartize params in Beacon-Chain and Validator Projects

move validator/params and beacon-chain/params to shared/params
instead of SetEnv define SetDemoBeaconConfig() that handles config parameters
for clarity rename functions that handle validator and beacon-chain configs.

### CODE REVIEW

File: shared/params/config.go [hunk 1]
```diff
@@ -111,27 +112,87 @@ const (
 )
 
 // GetConfig retrieves beacon node config.
```
Comment (terenc3t) [inline shared/params/config.go]: // GetBeaconConfig

File: shared/params/config.go [hunk 2]
```diff
@@ -111,27 +112,87 @@ const (
...
-	SetEnv("custom")
+	v := &OverrideConfig{}
+	v.param = "custom"
 	customConfig = c
 }
 
-// SetEnv sets which config to use.
-func SetEnv(e string) {
-	env = e
+// SetCustomConfig is useful for simulated backend configurations
```
Comment (terenc3t) [inline shared/params/config.go]: Delete commented code

File: shared/params/config.go [hunk 3]
```diff
@@ -111,27 +112,87 @@ const (
...
+// // SetEnv sets which config to use.
+// func SetEnv(e string) {
+// 	env = e
+// }
+
+////end beaconConfig
+
+
+///startt validator config
+// DefaultConfig returns pointer to a Config value with same defaults.
```
Comment (terenc3t) [inline shared/params/config.go]: s/Config/config

File: shared/params/config_test.go [hunk 4]
```diff
@@ -52,3 +55,12 @@ func TestValidatorSetDeltaFlags(t *testing.T) {
 		}
 	}
 }
+
+
+//validator params test
+func TestAttesterCommitteeSize(t *testing.T) {
```
Comment (terenc3t) [inline shared/params/config_test.go]: This has nothing to do with attester committee size. Perhaps TestCollationSize is more appropriate?

File: shared/params/config.go [hunk 5]
```diff
@@ -111,27 +112,87 @@ const (
...
+	}
+}
+
+// DefaultCollationSizeLimit is the integer value representing the maximum
+// number of bytes allowed in a given collation.
+func DefaultCollationSizeLimit() int64 {
+	return int64(math.Pow(float64(2), float64(20)))
 }
+
+// // Config contains configs for node to participate in the sharded universe.
```
Comment (terenc3t) [inline shared/params/config.go]: please delete commented code

File: shared/params/config.go [hunk 6]
```diff
@@ -111,27 +112,87 @@ const (
 )
 
 // GetConfig retrieves beacon node config.
-func GetConfig() *Config {
+func GetBeaconConfig() *Config {
+	var oc *OverrideConfig
+  	if oc != nil && oc.param != ""{
+	env = oc.param
```
Comment (terenc3t) [inline shared/params/config.go]: Are you sure this works? Can we test this to make sure we can get "demo"?

File: shared/params/config.go [hunk 7]
```diff
@@ -111,27 +112,87 @@ const (
...
+// func SetEnv(e string) {
+// 	env = e
+// }
+
+////end beaconConfig
+
+
+///startt validator config
+// DefaultConfig returns pointer to a Config value with same defaults.
+func DefaultConfig() *Config {
```
Comment (terenc3t) [inline shared/params/config.go]: This should be `DefaultValidatorConfig()`

**Comment by rauljordan** [comment]
Hi @cyberbono3 any update on this?

File: beacon-chain/types/crystallized_state.go [hunk 8]
```diff
@@ -283,20 +283,23 @@ func (c *CrystallizedState) NewStateRecalculations(aState *ActiveState, block *B
 	recentBlockHashes := aState.RecentBlockHashes()
 	newState.data.Validators = casper.CopyValidators(newState.Validators())
 
-	if c.LastStateRecalculationSlot() < params.GetConfig().CycleLength {
+	if c.LastStateRecalculationSlot() < params.GetBeaconConfig().CycleLength {
 		lastStateRecalculationSlotCycleBack = 0
 	} else {
-		lastStateRecalculationSlotCycleBack = c.LastStateRecalculationSlot() - params.GetConfig().CycleLength
+		lastStateRecalculationSlotCycleBack = c.LastStateRecalculationSlot() - params.GetBeaconConfig().CycleLength
+<<<<<<< HEAD
```
Comment (prestonvanloon) [inline beacon-chain/types/crystallized_state.go]: Merge conflicts here

File: shared/params/config.go [hunk 9]
```diff
@@ -110,28 +111,84 @@ const (
 	Exit
 )
 
-// GetConfig retrieves beacon node config.
-func GetConfig() *Config {
+<<<<<<< HEAD:shared/params/config.go
```
Comment (prestonvanloon) [inline shared/params/config.go]: merge conflict

**Comment by prestonvanloon** [comment]
Superseded by #796

**Pull request closed without merging**